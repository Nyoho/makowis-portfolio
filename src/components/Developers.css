:local .developers > :not(:first-child) {
    margin: 0 0 0 10px;
}
