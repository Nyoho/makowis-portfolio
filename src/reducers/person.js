// @flow
import person from '../models/Person';
import type { Person } from '../models/Person';

const CHANGE_PERSON: 'CHANGE_PERSON' = 'CHANGE_PERSON';

export type State = Person;

type Action = { type: typeof CHANGE_PERSON, person: Person };

const reducer = (state: Person = person, action: Action): State => {
  switch (action.type) {
    case CHANGE_PERSON:
      return action.person;
    default:
      return state;
  }
};

export default reducer;
