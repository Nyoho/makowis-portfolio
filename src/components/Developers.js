// @flow
import React from 'react';
import Avatar from './Avatar';
import { developers as className } from './Developers.css';
import type { Person } from '../models/Person';

const AvatarForPerson = ({ avatar, name, ...rest }: Person) => (
  <Avatar {...{ ...rest, src: avatar, alt: name }} />
);

type Props = {
  people: Person[],
  onPersonClick: (Person) => (SyntheticTouchEvent<any>) => void,
};

const Developers = ({ people, onPersonClick }: Props) => (
  <div {...{ className }}>
    {people.map((person) => (
      <AvatarForPerson
        {...person}
        key={person.id}
        onClick={onPersonClick(person)}
        width={40}
        height={40}
      />
    ))}
  </div>
);

export default Developers;
