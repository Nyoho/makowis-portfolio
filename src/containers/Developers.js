/* eslint-env browser */
// @flow
import { connect } from 'react-redux';

import type { Dispatch } from 'redux';

import Component from '../components/Developers';
import { developers } from '../models/Person';
import { changePerson } from '../reducers/person';

const mapStateToProps = () => ({
  people: developers,
});

// noinspection JSUnusedGlobalSymbols
const mapDispatchToProps = (dispatch: Dispatch) => ({
  // なぜか IDEA に Unusedっていわれる。
  onPersonClick: (person) => () => {
    // TODO react-router を導入すれば削除できるので、直接扱う
    window.history.pushState(null, person.name, `/${person.id}`);
    dispatch(changePerson(person));
  },
});

const Developers = connect(mapStateToProps, mapDispatchToProps)(Component);

export default Developers;
