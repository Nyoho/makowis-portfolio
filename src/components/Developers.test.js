/* eslint-env jest */
// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Developers from './Developers';
import person from '../models/Person';

const mockPersonClick = () => () => {};

it('renders Developers', () => {
  const tree = renderer
    .create(<Developers people={[person]} onPersonClick={mockPersonClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
