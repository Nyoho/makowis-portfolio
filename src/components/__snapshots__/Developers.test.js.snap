// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Developers 1`] = `
<div>
  <button
    className="mdc-fab "
    disabled={false}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
  >
    <img
      alt="Makoto Henmi"
      facebook="makoto.henmi"
      github="makowis"
      height={60}
      id="mako_wis"
      nickname="mako_wis"
      onClick={[Function]}
      raised={true}
      speakerdeck="makowis"
      src="avatar.bmp"
      twitter="mako_wis"
      width={60}
    />
  </button>
</div>
`;
